### DISCUSSION TITLE
Smb driver gets Access is Denied error

### DISCUSSION THREAD

**Discussion by pnookala-px** [kubernetes-csi/csi-driver-smb#622] [Q&A] [answered]
Hi Team,
I have a custom CSI driver based on the csi driver above. I have a cluster comprising of 3 linux nodes and 1 storage node. The linux nodes provide the storage. Created a share on the linux machine, exported it over samba and then am trying to launch a pod using that. 

On the windows node, during nodestage volume.
1. Mounted the volume using new-smbmapping
2. Symlink a temporary placeholder to the mount location.
3. Symlink from the "globalmount" to temporary placeholder.

During node publish volume 
4. Symlink from "mount" to "globalmount".
Error: failed to generate container "cdd8d39a5a15ed1b5c949d4b5ce4579d0992a2fd324370c083a6d799614f62bd" spec: failed to generate spec: failed to stat "c:\\var\\lib\\kubelet\\pods\\cc2ce367-fa43-4514-9863-be74ee4a7847\\volumes\\kubernetes.io~csi\\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\\mount": CreateFile c:\var\lib\kubelet\pods\cc2ce367-fa43-4514-9863-be74ee4a7847\volumes\kubernetes.io~csi\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\mount: Access is denied.
All these calls return successfully and then during bindmount
the calls fail with the error':

{\"message\":\"failed to generate container \\\"f5f006e6ec08126a8a7d0d3db777a39bfb95da58f2730ce08f4ecaf327ab8d27\\\" spec: failed to generate spec: failed to stat \\\"c:\\\\\\\\var\\\\\\\\lib\\\\\\\\kubelet\\\\\\\\pods\\\\\\\\cc2ce367-fa43-4514-9863-be74ee4a7847\\\\\\\\volumes\\\\\\\\kubernetes.io~csi\\\\\\\\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\\\\\\\\mount\\\": CreateFile c:\\\\var\\\\lib\\\\kubelet\\\\pods\\\\cc2ce367-fa43-4514-9863-be74ee4a7847\\\\volumes\\\\kubernetes.io~csi\\\\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\\\\mount: Access is denied.\",\"reason\":\"CreateContainerError\"}}},{\"containerID\":\"containerd://0e81eb630f5478baaf524a91193585a22496639993c9dd8753b5cf08e923a0fb\",\"image\":\"mcr.microsoft.com/windows/servercore:ltsc2019\",\"imageID\":\"mcr.microsoft.com/windows/servercore@sha256:17bf5541b81ad3cfb8eb5170c2ce4f06df0d3741bb573fe27c7b5729d9c010f4\",\"lastState\":{},\"name\":\"windowswebserver\",\"ready\":true,\"restartCount\":0,\"started\":true,\"state\":{\"running\":{\"startedAt\":\"2023-05-18T00:47:58Z\"}}}],\"podIP\":\"10.100.49.153\",\"podIPs\":[{\"ip\":\"10.100.49.153\"}]}}"

Error: failed to generate container "cdd8d39a5a15ed1b5c949d4b5ce4579d0992a2fd324370c083a6d799614f62bd" spec: failed to generate spec: failed to stat "c:\\var\\lib\\kubelet\\pods\\cc2ce367-fa43-4514-9863-be74ee4a7847\\volumes\\kubernetes.io~csi\\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\\mount": CreateFile c:\var\lib\kubelet\pods\cc2ce367-fa43-4514-9863-be74ee4a7847\volumes\kubernetes.io~csi\pvc-aaa0c128-b5bc-41e6-b002-596cb8da2d1d\mount: Access is denied.

Has anyone seen this earlier?

**Comment by andyzhangx** [comment]
have you installed csi-proxy on the windows node?

**Reply by andyzhangx** [reply, marked as answer]
you should use `New-SmbGlobalMapping` to mount smb share instead of `new-smbmapping` since if you are using `new-smbmapping`,  you cannot fetch the smb credentials inside container while if you use  `New-SmbGlobalMapping`, it should work.

btw, why are you forking this smb csi driver?

**Comment by pnookala-px** [comment]
Yes.

**Comment by pnookala-px** [comment]
Thanks Andy.
Use New-SmbGlobalMapping worked. I am trying to play around with the driver
by making some modifications of my own.

regards,
Phani.
